src/config/config_impl.go

```diff
@@ -306,10 +306,19 @@ func (this *rateLimitConfigImpl) GetLimit(
 
 	descriptorsMap := value.descriptors
 	prevDescriptor := &value.rateLimitDescriptor
+
+	// Build detailed metric as we traverse the list of descriptors
+	var detailedMetricFullKey strings.Builder
+	detailedMetricFullKey.WriteString(domain)
```

Comment:
perf nit: can we only do this if detailed metrics are enabled?

Reply:
i fear it's not possible. only once we traversed the "tree", we know if the leaf is a detailed metric (it can be hierarchical, and the leaf is the deciding factor).

so we'd need to build this string, even if we discard it later (line 364).

the previous solution loses this information because of exactly this reason (keys of previous descriptors in the "tree" are lost).

i can think of two alternatives:
- we could traverse the tree again, after we have traversed it once, and see that the leaf has detailed metrics on (line 364). I haven't benchmarked, but i don't feel that this is faster than doing a bit of string concatenation
- add a settings option to ratelimit-service, that steers detailed metrics in general. i.e. you can completely turn it off if you wish. however, even here we would spend a little more time doing this string concat, if somebody wants to have detailed metrics only for a small subset of their metrics.

Reply:
Sure fair enough, thanks for the detailed explanation. SGTM.